buildtools_test.go

```diff
@@ -308,6 +311,64 @@ func TestGoBuildInfo(t *testing.T) {
 	cleanGoTest(gopath)
 }
 
+func TestGoWithConfig(t *testing.T) {
+	initGoTest(t)
+
+	oldHomeDir := os.Getenv(cliutils.JfrogHomeDirEnv)
+	defer os.Setenv(cliutils.JfrogHomeDirEnv, oldHomeDir)
+	// Populate cli config with 'default' server
+	createJfrogHomeConfig(t)
+	wd, err := os.Getwd()
+	if err != nil {
+		t.Error(err)
+	}
+	gopath := os.Getenv("GOPATH")
+	os.Setenv("GOPATH", filepath.Join(wd, tests.Out))
+	project1Path := createGoProject(t, "project1", true)
+	testsdataTarget := filepath.Join(tests.Out, "testsdata")
+	testsdataSrc := filepath.Join(filepath.FromSlash(tests.GetTestResourcesPath()), "go", "testsdata")
+	err = fileutils.CopyDir(testsdataSrc, testsdataTarget, true)
+	if err != nil {
+		t.Error(err)
+	}
+	err = os.Chdir(project1Path)
+	if err != nil {
+		t.Error(err)
+	}
+	defer os.Chdir(wd)
+
+	log.Info("Using Go project located at ", project1Path)
+
+	buildName := "go-build"
+
+	// 1. Download dependencies.
+	// 2. Publish build-info.
+	// 3. Validate the total count of dependencies added to the build-info.
+	buildNumber := "1"
+	// Preparing config file.
+	configFileDir := filepath.Join(project1Path, ".jfrog", "projects")
+	configFileDir, err = tests.ReplaceTemplateVariables(filepath.Join(configFileDir, "go.yaml"), configFileDir)
+	if err != nil {
+		t.Error(err)
+	}
+	artifactoryGoCli := tests.NewJfrogCli(execMain, "jfrog rt", "")
```

Comment:
What is the reason for creating a new JFrog CLI here, instead of using the existing one?

Reply:
Since the existing one always send the flags of --url, --user, --password. Those flags can't be used with rt go and config file.